cfn-resources/cloud-backup-restore-jobs/cmd/resource/resource.go

```diff
@@ -335,39 +364,121 @@ func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.P
 	}, nil
 }
 
-// function to check id restore job already exist
-func isRestoreJobExist(client *mongodbatlas.Client, currentModel *Model) (isExist bool) {
-	var restoreJobs *mongodbatlas.CloudProviderSnapshotRestoreJobs
-	var err error
-	clusterName := cast.ToString(currentModel.ClusterName)
-	instanceName := cast.ToString(currentModel.InstanceName)
-	projectID := cast.ToString(currentModel.ProjectId)
-	params := &mongodbatlas.ListOptions{
-		PageNum:      0,
-		ItemsPerPage: 100,
+func (model *Model) validateAsynchronousProperties() error {
+	if model.EnableSynchronousCreation != nil && *model.EnableSynchronousCreation {
+		if model.SynchronousCreationOptions == nil {
+			model.SynchronousCreationOptions = &SynchronousCreationOptions{}
+		}
+
+		if model.SynchronousCreationOptions.CallbackDelaySeconds == nil {
+			model.SynchronousCreationOptions.CallbackDelaySeconds = aws.Int(defaultBackSeconds)
+		}
+
+		if model.SynchronousCreationOptions.TimeOutInSeconds == nil {
+			model.SynchronousCreationOptions.TimeOutInSeconds = aws.Int(defaultTimeOutInSeconds)
+		}
+
+		if model.SynchronousCreationOptions.ReturnSuccessIfTimeOut == nil {
+			model.SynchronousCreationOptions.ReturnSuccessIfTimeOut = aws.Bool(defaultReturnSuccessIfTimeOut)
+		}
 	}
 
-	if clusterName != "" {
-		snapshotRequest := &mongodbatlas.SnapshotReqPathParameters{
-			GroupID:     projectID,
-			ClusterName: clusterName,
+	return nil
+}
+
+func flowIsAsynchronous(model *Model) bool {
```

Comment:
Should this be called `flowIsSynchronous ` instead or am I misunderstanding? Because we check `*model.EnableSynchronousCreation` to be true

Reply:
yes wrong name